cypress/integration/registrationFlows/emailRegistration.spec.js

```diff
@@ -0,0 +1,27 @@
+describe('Sign up with email', () => {
+  beforeEach(() => {
+    cy.testSetup();
+    cy.visit('/enter?state=new-user');
+  });
+
+  // Regression test for #11099
+  it('should preserve the form data on unsuccesful submissions', () => {
+    cy.get('a').contains('Sign up with Email').click();
```

Comment:
It's great we can add this test - I mentioned one helper in the commands file, but we could make use of some other testing-library methods here too.

```suggestion
    cy.findByRole('link', {name: 'Sign up with Email'}).click();
```